### PR TITLE
fix: take into account security at form level

### PR DESCRIPTION
**Pull request by danielpeintner**
fixes https://github.com/eclipse/thingweb.node-wot/issues/466

### CODE REVIEW

**Comment by danielpeintner** [comment]
@relu91 @egekorkan a proper test is missing.

Having said that, we do not seem to have a good set of security tests where I could add one. Did I oversee something (and you can point me to it).. since I am somewhat in a hurry 🙈

File: packages/core/src/consumed-thing.ts [hunk 1]
```diff
@@ -121,19 +121,30 @@ export default class ConsumedThing extends TD.Thing implements WoT.ConsumedThing
...
-                let ws = this.securityDefinitions[s + ""]; // String vs. string (fix wot-typescript-definitions?)
-                // also push nosec in case of proxy
-                if (ws) {
-                    scs.push(ws);
-                }
+
+    getSecuritySchemes(security: Array<String>): Array<TD.SecurityScheme> {
+        let scs: Array<TD.SecurityScheme> = [];
+        for (let s of security) {
+            let ws = this.securityDefinitions[s + ""]; // String vs. string (fix wot-typescript-definitions?)
```
Comment (relu91) [inline packages/core/src/consumed-thing.ts:128]: Is this comment still valid? 🤔

**Review by relu91 — APPROVED** [review]
Changes look good. I agree that we would need a proper unit test for this feature. I don't recall to have found any. In `binding-http` there are just some tests about credentials and OAuth2.0 flows.

File: packages/core/src/consumed-thing.ts (hunk 1 above)
Comment (danielpeintner) [reply packages/core/src/consumed-thing.ts:128]: What is strange is the following:
security on thing level is of type: `security: Array<String>`
security on form level is of type: `security?: Array<string>`
`String` vs. `string`

**Comment by danielpeintner** [comment]
> I agree that we would need a proper unit test for this feature.

I added 2 very simple tests.

File: packages/core/src/consumed-thing.ts (hunk 1 above)
Comment (relu91) [reply packages/core/src/consumed-thing.ts:128]: I checked the new TD types and I don't see this problem. We should remember to remove the comment once we merge with 0.8.x

**Review by relu91 — APPROVED** [review]
Good to go!

**Pull request merged**